fabric-object-builders-v0/src/main/java/net/fabricmc/fabric/impl/object/builder/TradeOfferFactories.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.object.builder;
+
+import net.minecraft.enchantment.EnchantmentHelper;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemConvertible;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.village.TradeOffer;
+import net.minecraft.village.TradeOffers;
+
+public final class TradeOfferFactories {
+	@FunctionalInterface
+	public interface SellEnchantedTool {
+		SellEnchantedTool FACTORY = (item, basePrice, maxUses, experience, multiplier) -> ((entity, random) -> {
+			int level = 5 + random.nextInt(15);
```

Comment:
Im not a fan of having this logic here

Reply:
There is a reason behind this is because the builtin enchanted tool factory is a package private inner class. 

It's basically the same logic as the default villager trades factory but if it isn't 100% needed (since anyone can create a TradeOffer factory) we could remove it or handle this in another way.

Reply:
With the way it is, if the vanilla logic changes this will require updating, and chances are that wont happen.

Reply:
One option I have is to use a bunch of reflection hacks to refer to the vanilla factories (since ATs are not an option) but a rename of a class file or change to the ctors will obviously break it there. Also it's just plain nasty code for another 3-4 trade types which are are quite dependent on a specific profession [And the type aware trade factory is straight up hardcoded with VillagerTypes].

I probably will just leave it to the current three generic trade offer types and take out the enchanted item since a mod dev can add their own trade factories.